packages/dd-trace/test/appsec/iast/analyzers/template-injection-analyzer.handlebars.plugin.spec.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const { prepareTestServerForIast } = require('../utils')
+const { storage } = require('../../../../../datadog-core')
+const iastContextFunctions = require('../../../../src/appsec/iast/iast-context')
+const { newTaintedString } = require('../../../../src/appsec/iast/taint-tracking/operations')
+
+describe('template-injection-analyzer with handlebars', () => {
+  withVersions('handlebars', 'handlebars', version => {
+    let lib, source
+    before(() => {
+      lib = require(`../../../../../../versions/handlebars@${version}`).get()
+      source = '<p>{{name}}</p>'
+    })
+
+    describe('compile', () => {
+      prepareTestServerForIast('template injection analyzer',
+        (testThatRequestHasVulnerability, testThatRequestHasNoVulnerability) => {
+          testThatRequestHasVulnerability(() => {
+            const store = storage.getStore()
+            const iastContext = iastContextFunctions.getIastContext(store)
+            const template = newTaintedString(iastContext, source, 'param', 'Request')
+            lib.compile(template)
+          }, 'TEMPLATE_INJECTION')
+
+          testThatRequestHasNoVulnerability(() => {
+            const template = lib.compile(source)
+            template()
```

Comment:
`template()` is invoked in this test (no vuln), but is not invoked in the previous (when checking there is a vuln). Is there any reason for that?